tests/test_integration.py

```diff
@@ -1,6 +1,63 @@
+import sys
+import os
 
+# Set the path to the src directory
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../src')))
+from src import model_definition
+from src import preprocess
+from src import predict
+from src import train
+import numpy as np
 import pytest
+import yaml
+import json
 
```

Comment:
About the markers. The pipeline is hardcoded on the markers (which I guess we could have a file to keep track of them all or maybe pytest provides some logic for). But anyhow, I kinda liked unit and integration, but also like fast and slow. The question is, is slow meant only for integration or also for other kind of tests we may add later? What defines speed? 

Previously, the integration tests were only ran when PR is open with main. Unittest are ran for any push and PR. The question is, do we want the same thing for fast and slow, or will we have some slow ones on every push? 

In general, I think having specific markers is better, if it doesn't get out of hand.

Reply:
I think we could also have markers based on the 4 angles described in the lecture.

Reply:
Yeah that would probably work better with the grading scheme as well.